librosa/core/pitch.py

```diff
@@ -389,31 +384,25 @@ def _cumulative_mean_normalized_difference(
     yin_frames : np.ndarray [shape=(max_period-min_period+1,n_frames)]
         Cumulative mean normalized difference function for each frame.
     """
-    # Autocorrelation.
-    a = np.fft.rfft(y_frames, frame_length, axis=-2)
-    b = np.fft.rfft(y_frames[..., win_length:0:-1, :], frame_length, axis=-2)
-    acf_frames = np.fft.irfft(a * b, frame_length, axis=-2)[..., win_length:, :]
-    acf_frames[np.abs(acf_frames) < 1e-6] = 0
+    acf_frames = audio.autocorrelate(y_frames, max_size=max_period + 1, axis=-2)
 
     # Energy terms.
-    energy_frames = np.cumsum(y_frames**2, axis=-2)
-    energy_frames = (
-        energy_frames[..., win_length:, :] - energy_frames[..., :-win_length, :]
-    )
-    energy_frames[np.abs(energy_frames) < 1e-6] = 0
+    yin_frames = np.cumsum(np.square(y_frames), axis=-2)
 
-    # Difference function.
-    yin_frames = energy_frames[..., :1, :] + energy_frames - 2 * acf_frames
+    # Difference function: d(k) = 2 * (ACF(0) - ACF(k)) - sum_{m=0}^{k-1} y(m)^2
+    k = slice(1, max_period + 1)
+    yin_frames[..., 0, :] = 0
+    yin_frames[..., k, :] = (
+        2 * (acf_frames[..., 0, :] - acf_frames[..., k, :]) - yin_frames[..., :k.stop-1, :]
+    )
 
     # Cumulative mean normalized difference function.
     yin_numerator = yin_frames[..., min_period : max_period + 1, :]
     # broadcast this shape to have leading ones
-    tau_range = util.expand_to(
-        np.arange(1, max_period + 1), ndim=yin_frames.ndim, axes=-2
-    )
+    k_range = util.expand_to(np.r_[k], ndim=yin_frames.ndim, axes=-2)
 
     cumulative_mean = (
-        np.cumsum(yin_frames[..., 1 : max_period + 1, :], axis=-2) / tau_range
+        np.cumsum(yin_frames[..., k, :], axis=-2) / k_range
```

Comment:
somewhere in this passage, I think, is where the multichannel failures are creeping in.  @dsuedholt could you step this on a multi-channel example and make sure that everything retains the proper shapes compared to the current main branch?